pywbem_mock/_wbemconnection_mock.py

```diff
@@ -995,6 +1012,104 @@ def _display_objects(obj_type, object_repo, namespace, cmt_begin, cmt_end,
                     else:
                         _uprint(dest, obj.tomof())
 
+    def register_provider(self, namespaces, classnames, provider_type,
+                          provider):
+        """
+        Register the provider object for namespace and class. Registering a
+        provider tells the FakedWBEMConnection that the provider implementation
+        provided with this call is to be executed as the request response
+        method for the namespace and class defined in lieu of the default
+        provider.
+
+        Providers can only be registered for request response methods
+        CreateInstance and DeleteInstance and InvokeMethod
+
+        Parameters:
+
+          namespace (:term:`string or :class:`py:list` of :term:`string):
+            Namespace or namespaces for which the provider is being registered.
+            If empty or None, the classname is registered for all namespaces
+
+          classname (:term:`string or :class:`py:list` of :term:`string):
+            Classname or classnames for which the provider is being
+            registered. At least one classname is required
+
+          # TODO Requirement is that it is a class that is a subclass of
+          # DefaultInstanceProvider
+          provider (:class:):
+            The user provider class which is a subclass of
+            DefaultInstanceProvider.  The methods in this subclass override
+            the corresponding methods in DefaultInstanceProvider.
+            The method call parameters must be the same as the defult method in
+            DefaultInstanceProvider and it must return data in the same format
+            if the default method returns data.
+        """
+        provider_types = ('instance', 'method')
+
+        if provider_type not in provider_types:
+            raise ValueError("{0!A} is not a valid provider type. "
+                             "Valid provider types are {1!A}.", provider_type,
+                             provider_types)
+
+        if not isinstance(provider, DefaultInstanceProvider):
+            raise ValueError(
+                _format("{0!A} is not a valid subclass of "
+                        "DefaultInstanceProvider. ", provider))
+
+        if classnames is None:
+            raise ValueError(
+                _format('register_provider classnames argument '
+                        'must be string or list '
+                        'of strings. None not allowed.'))
```

Comment:
Instead of just excluding None, I think this should test for the allowed types, and raise TypeError.

Reply:
We do that in second test in the recursion with each classname.  However, This initial test is needed since we do not want to do the recursion with classname None.

**DISCUSSION**

Reply:
Invalid type. string or list of strings allowed as message.

Check test should be string or list.

TODO

Reply:
Was not changed. Marking the comment as resolved - this will be reviewed further.